websites/S/Streamlabs/presence.ts

```diff
@@ -1,65 +1,172 @@
+
+				if (path[2] === "post") {
+					postTitle = document
+						.querySelector(".content-card__title")
+						.firstChild.textContent.trim();
+					presenceData.state = `${postTitle}`;
+
+					presenceData.buttons = [
+						{
+							label: "View Post",
+							url: document.location.href
+						},
+						{
+							label: "View Author",
+							url: (
+								document.querySelector(
+									".content-card__author-link"
+								) as HTMLAnchorElement
+							).href
+						}
+					];
+				}
 				break;
-			case "/best-donation-clips":
+
+			case "dashboard":
+				switch (hash[1]) {
+					case "":
+						titleDashboard = "Viewing their Dashboard";
+						break;
+
+					case "stats":
+						titleDashboard = "Viewing their Stats";
+						break;
+
+					case "advancedstats":
+						titleDashboard = "Viewing their Stats";
+						break;
+
+					case "recentevents":
+						titleDashboard = "Viewing their Recent Events";
+						break;
+
+					case "alertbox":
+						titleDashboard = "Viewing their Alert Box";
+						break;
+
+					case "widgets":
+						titleDashboard = "Viewing their Widgets";
+						break;
+
+					case "cloudbot":
+						titleDashboard = "Viewing their Cloudbot";
+						break;
+
+					case "multistream":
+						titleDashboard = "Viewing their MultiStream";
+						break;
+
+					case "prime":
+						switch (hash[2]) {
+							case "panels":
+								titleDashboard = "Viewing their Panels";
+								break;
+
+							case "thumbnails":
+								titleDashboard = "Viewing their Thumbnails";
+								break;
+						}
+						break;
+
+					case "logo-maker":
+						titleDashboard = "Creating their Logo";
+						break;
+
+					case "intro-maker":
+						titleDashboard = "Creating their Intro";
+						break;
+
+					case "merchadmin":
+						titleDashboard = "Creating their Merch";
+						break;
+
+					case "social-reminders":
+						titleDashboard = "Creating their Social Reminders";
+						break;
+
+					case "settings":
+						titleDashboard = "Viewing their Settings";
+						break;
+
+					case "monthly-tips":
+						titleDashboard = "Creating their Monthly Tips";
+						break;
+
+					case "donations":
+						titleDashboard = "Viewing their Donations";
+						break;
+
+					case "subscribers":
+						titleDashboard = "Viewing their Subscribers";
+						break;
+					case "streamlabs-rewards":
+						titleDashboard = "Viewing their Rewards";
+						rewardString = document.querySelector(
+							"#sl__dashboard > div > div.content > div.dashboard-content > div.dashboard-body > div > div > div:nth-child(1) > div:nth-child(1) > span"
+						).textContent;
+						presenceData.state =
+							rewardString.charAt(0).toUpperCase() + rewardString.slice(1);
+						//Streamlabs rewards does not captilise first string of the teir.
+						break;
+				}
 				presenceData.startTimestamp = browsingTimestamp;
-				presenceData.details = `${userType[0]}Best Clips`;
+				presenceData.details = titleDashboard;
```

Comment:
Is titleDashboard used past this point? You could just assign these values to presenceData.details directly if not